### ISSUE TITLE
[rllib] Parametric actions cartpole example with DQN: No invalid actions during exploration

### ISSUE THREAD

**Issue by klausk55** [ray-project/ray#9191]
I think the parametric actions cartpole example running with DQN uses by default "EpsilonGreedy" for exploration.
How and where is it controlled that during exploration no invalid actions are chosen?

When not exploring, then the action mask give invalid action logits no mass.

**Comment by klausk55** [comment]
I think I found it in the `EpsilonGreedy` class:
```
# Mask out actions with q-value=-inf so that we don't even consider
# them for exploration.
random_valid_action_logits = tf.where(
            tf.equal(q_values, tf.float32.min),
            tf.ones_like(q_values) * tf.float32.min, tf.ones_like(q_values))
random_actions = tf.squeeze(
            tf.multinomial(random_valid_action_logits, 1), axis=1)
```
